File: _CodeTopics/LeetCode/1601-1800/001720/001720.py
class Solution(object):
    def decode(self, encoded, first):
        """
        :type encoded: List[int]
        :type first: int
        :rtype: List[int]
        """

        res = [first]
        for elem in encoded:
            res.append(elem^res[-1])
        return res
        
"""
https://leetcode-cn.com/submissions/detail/174886292/

76 / 76 个通过测试用例
状态：通过
执行用时: 44 ms
内存消耗: 14.5 MB

执行用时：44 ms, 在所有 Python 提交中击败了68.90%的用户
内存消耗：14.5 MB, 在所有 Python 提交中击败了60.77%的用户
"""

File: _CodeTopics/LeetCode/1601-1800/001720/tran/001720.cpp
class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int> res = {first};
        for (auto elem : encoded) {
            res.push_back(elem ^ res.back());
        }
        return res;
    }
};

/*
https://leetcode-cn.com/submissions/detail/175232525/

76 / 76 个通过测试用例
状态：通过
执行用时: 32 ms
内存消耗: 25.2 MB

执行用时：32 ms, 在所有 C++ 提交中击败了71.27%的用户
内存消耗：25.2 MB, 在所有 C++ 提交中击败了47.84%的用户
*/
